hypertrace-trace-enricher/trace-reader/src/main/java/org/hypertrace/trace/reader/attributes/ValueCoercer.java

```diff
@@ -1,12 +1,100 @@
 package org.hypertrace.trace.reader.attributes;
 
+import static java.util.Objects.requireNonNull;
+
+import java.time.Instant;
 import java.util.Optional;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import org.hypertrace.core.attribute.service.v1.AttributeKind;
 import org.hypertrace.core.attribute.service.v1.LiteralValue;
 
-interface ValueCoercer {
+public class ValueCoercer {
```

Comment:
Any changes besides inlining the implementation? Also same comment about making this not constructable. Not sure if that's a java pattern or not, but I don't see any downsides to using an interface over a class? I guess making it final with a private constructor accomplishes the same goal though.

Reply:
> Any changes besides inlining the implementation

no changes